docs/ldml/tr35-keyboards.md

```diff
@@ -2908,6 +2908,196 @@ This attribute specifies the expected resultant text in a document after process
     <check result="abc\u{0022}s\u{0022}•éÈ"/>
 </test>
 ```
+## Appendix: Pseudocode for Marker-safe Normalization
+
+The following is an informative section with some example pseudocode for marker-safe normalization into NFD.
+
+1. Markers are not text. This pseudocode assumes an implementation which is able to distinguish markers from plain text.
+
+2. Some optimization notes are given, but the emphasis here is on a clear and correct implementation.
+
+3. A distinction would need to be made between (A) markers in text, such as a keystroke emitting `\m{someMarker}` or a marker in input context, and (B) markers in transform match strings, such as `<transform from="\m{marker}">` or `<transform from="\m{.}">`.  An implementation could pass a parameter through the following code path distinguishing A and B, this distinction is not shown, for clarity.
+
+Example use:
+
+- `nfdMarkers('e\u{0300}' + Marker('marker1') + '\u{0320}a\u{0300}' + Marker('marker2') + '\u{0320}')`
+- Returns: `'e' + Marker('marker1') + '\u{0320}\u{0300}a' + Marker('marker2') + '\u{0320}\u{0300}'`
```

Comment:
Reminder of a point from the original PR: https://github.com/unicode-org/cldr/pull/3445#discussion_r1470374770

> > Since you use `'\u{0300}'` and `'\u{0320}'` following both the `'e'` and the `'a'`, it's easy to confuse what's happening in the center part.
>
> I agree with the comment. Don't use duplicate characters in the example when the positions may change.